knotx-core/src/main/java/io/knotx/launcher/KnotxStarterVerticle.java

```diff
@@ -54,48 +52,64 @@ public void start(Future<Void> startFuture) throws Exception {
         );
   }
 
-  private Observable<Pair<String, String>> verticleCouldNotBeDeployed(Object module,
-      Throwable throwable) {
-    LOGGER.warn("Can't deploy {}: {}", module, throwable.getMessage());
-    return Observable.empty();
-  }
-
-  private Observable<Pair<String, String>> deployVerticle(final Object module) {
-    return vertx.rxDeployVerticle((String) module, getModuleOptions((String) module))
-        .map(deploymentID -> Pair.of((String) module, deploymentID))
+  private Observable<ModuleDescriptor> deployVerticle(final ModuleDescriptor module) {
+    return vertx
+        .rxDeployVerticle(module.getName(), getModuleOptions(module.getAlias()))
+        .map(new ModuleDescriptor(module)::setDeploymentId)
+        .doOnError(error -> LOGGER.error("Can't deploy {}: {}", module.toDescriptorLine(), error))
+        .onErrorResumeNext((err) -> Single.just(new ModuleDescriptor(module).setDeploymentFailed()))
         .toObservable();
   }
 
   private DeploymentOptions getModuleOptions(final String module) {
     DeploymentOptions deploymentOptions = new DeploymentOptions();
-    if (config().containsKey(CONFIG_OVERRIDE) && config().getJsonObject(CONFIG_OVERRIDE)
-        .containsKey(module)) {
-      JsonObject moduleConfig = config().getJsonObject(CONFIG_OVERRIDE).getJsonObject(module);
-      if (moduleConfig.containsKey(MODULE_OPTIONS)) {
-        deploymentOptions.fromJson(moduleConfig.getJsonObject(MODULE_OPTIONS));
+    if (config().containsKey(CONFIG_OVERRIDE)) {
+      if (config().getJsonObject(CONFIG_OVERRIDE).containsKey(module)) {
+        JsonObject moduleConfig = config().getJsonObject(CONFIG_OVERRIDE).getJsonObject(module);
+        if (moduleConfig.containsKey(MODULE_OPTIONS)) {
+          deploymentOptions.fromJson(moduleConfig.getJsonObject(MODULE_OPTIONS));
```

Comment:
From "options"? Not "deploymentOptions" ?

Reply:
yes, options is the name of the object. it always be, it's because the vertx expect to have it on "options"